types/Address/Address.d.ts

```diff
@@ -0,0 +1,142 @@
+import { IObjectWithId } from '../base';
+import { DeepPartial } from '../utils';
+import { IAddressCreateParameters } from './AddressCreateParameters';
+import { IVerifications } from './Verifications';
+
+/**
+ * Address objects are used to represent people, places, and organizations in a number of contexts.
+ * For example, a Shipment requires a to_address and from_address to accurately calculate rates and generate postage.
+ *
+ * Additionally, EasyPost offers several verification tools that can be used to detect deliverability issues, correct minor errors in spelling/formatting, and determine if an Address is residential or not (which has a significant effect on Shipment rating for many carriers).
+ *
+ * @see https://www.easypost.com/docs/api/node#address-object
+ */
+export declare interface IAddress extends IObjectWithId<'Address'> {
+  /**
+   * First line of the address
+   */
+  street1: string;
+
+  /**
+   * Second line of the address
+   */
+  street2: string;
+
+  /**
+   * City the address is located in
+   */
+  city: string;
+
+  /**
+   * State or province the address is located in
+   */
+  state: string;
+
+  /**
+   * ZIP or postal code the address is located in
+   */
+  zip: string;
+
+  /**
+   * ISO 3166 country code for the country the address is located in
+   */
+  country: string;
+
+  /**
+   * Whether or not this address would be considered residential
+   */
+  residential: boolean;
+
+  /**
+   * The specific designation for the address (only relevant if the address is a carrier facility)
+   */
+  carrier_facility: string;
+
+  /**
+   * Name of the person. Both name and company can be included
+   */
+  name: string;
+
+  /**
+   * Name of the organization. Both name and company can be included
+   */
+  company: string;
+
+  /**
+   * Phone number to reach the person or organization
+   */
+  phone: string;
+
+  /**
+   * Email to reach the person or organization
+   */
+  email: string;
+
+  /**
+   * Federal tax identifier of the person or organization
+   */
+  federal_tax_id: string;
+
+  /**
+   * State tax identifier of the person or organization
+   */
+  state_tax_id: string;
+
+  /**
+   * The result of any verifications requested
+   */
+  verifications: IVerifications;
+}
+
+export declare class Address implements IAddress {
+  public constructor(input: DeepPartial<IAddressCreateParameters>);
+
+  street1: string;
```

Comment:
What's the purpose of redefining all these properties here and above? Is there not a way to document each property at the same time we define it with the type? Seems like a lot of duplicate code.